src/distilabel/tasks/text_generation/evol_instruct.py

```diff
@@ -0,0 +1,193 @@
+        return ["instruction"]
+
+    def _elimination_evolving(self, output: str) -> Optional[str]:
+        """Performs the elimination step of the Evol-Instruct task, steps 2-4 in the paper:
+
+        1. [NOT IMPLEMENTED] The evolved instruction does not provide any information gain compared
+        to the original one. Use ChatGPT to make this determination, this is outlined in Appendix G of the original paper.
+        2. The evolved instruction makes it difficult for the LLM to generate a response. We found that
+        when the generated response contains “sorry” and is relatively short in length (i.e., less than
+        80 words), it often indicates that the LLM struggles to respond to the evolved instruction.
+        So we can use this rule to make a judgment.
+        3. The response generated by the LLM only contains punctuation and stop words.
+        4. The evolved instruction obviously copies some words from the evolving prompt, such as
+        “given prompt”, “rewritten prompt”, “#Rewritten Prompt#”, etc.
+        """
+        output = output.strip()
+        if output == "":
+            return
+
+        # 2) The evolved instruction makes it difficult for the LLM to generate a response.
+        if "sorry" in output.lower() and len(output.split(" ")) < 80:
+            logger.info(
+                f"Evolution step removed the output, it's hard for the LLM to generate a response: {output}"
+            )
+            return
+
+        # 3) The output only contains punctuation and stop words
+        stopwords = _get_stopwords()
+        clean_output = [word for word in output.split(" ") if word not in stopwords]
+        if set(clean_output).difference(set(string.punctuation)) == 0:
+            logger.info(
+                f"Evolution step removed the output, it only contains punctuation and stop words: {output}"
+            )
+            return
+
+        # 4) Remove copied words from the prompt
+        prompt_words = {
+            "#Given Prompt#",
+            "#Created Prompt#",
+            "given prompt",
+            "created prompt",
+            "#The Given Prompt#",
+            "#Rewritten Prompt#",
+            "rewritten prompt",
+        }
+        if any(word in output for word in prompt_words):
+            logger.info(
+                f"Evolution step removed the output due to word repetition from the prompt: {output}"
+            )
+            return
+
+        return output
+
+    def parse_output(self, output: str) -> Dict[str, List[str]]:
+        """Parses the output of the model into the desired format, applying the elimination step for bad generations.
+
+        Args:
+            output (str): the output of the model.
+
+        Note:
+            The eliminatin step is applied to the output, but only steps 2-4 in the paper are implemented.
+            Refer to point 3.2, Elimination Evolving section in the original paper for more information on the
```

Comment:
3.2 of what? should we not mention the Arxiv reference? And perhaps also do that in the main docstring?

Reply:
Updated with the explicit reference to the paper